app/lib/service/download_counts/computations.dart

```diff
@@ -42,3 +42,29 @@ Future<void> upload30DaysTotal(Map<String, int> counts) async {
   await uploadBytesWithRetry(reportsBucket, downloadCounts30DaysTotalsFileName,
       jsonUtf8Encoder.convert(counts));
 }
+
+Future<({List<int> weeklyPoints, DateTime? newestDate})> computeWeeklyDownloads(
+    String package) async {
+  final weeklyPoints = List.filled(52, 0);
+  final countData =
+      await downloadCountsBackend.lookupDownloadCountData(package);
+  if (countData == null) {
+    return (weeklyPoints: <int>[], newestDate: null);
+  }
+
+  final totals = countData.totalCounts;
+
+  var sum = 0;
+  for (int i = 0; i < 52 * 7; i++) {
```

Comment:
This is probably fine, but math would be easier to read if it was:
```suggestion
  for (int w = 0; w < 52; w++) {
    var sum = 0;
    for (int d = 0; d < 7; d++) {
      sum += totals[w * 7 + d];
    }
    weeklyPoints[w] = sum;
  }
```
Or maybe I'm missing something that it's supposed to do.

Or if one wants to be fancy :rofl: 
```suggestion
  weeklyPoints = List.generate(52, (w) => totals.skip(w * 7).take(7).sum);
```

Reply:
`List.generate(52, (w) => totals.skip(w * 7).take(7).where((c) => c > 0).sum);`

Reply:
The nested loop is nice, the fancy I think is not the best. :-D